ra/ra.go

```diff
@@ -633,11 +633,23 @@ func (ra *RegistrationAuthorityImpl) checkInvalidAuthorizationLimit(ctx context.
 // checkNewOrdersPerAccountLimit enforces the rlPolicies `NewOrdersPerAccount`
 // rate limit. This rate limit ensures a client can not create more than the
 // specified threshold of new orders within the specified time window.
-func (ra *RegistrationAuthorityImpl) checkNewOrdersPerAccountLimit(ctx context.Context, acctID int64) error {
+func (ra *RegistrationAuthorityImpl) checkNewOrdersPerAccountLimit(ctx context.Context, acctID int64, names []string) error {
 	limit := ra.rlPolicies.NewOrdersPerAccount()
 	if !limit.Enabled() {
 		return nil
 	}
+
+	// Check if there is already an existing certificate for the exact name set we
+	// are issuing for. If so bypass the the newOrders limit.
+	exists, err := ra.SA.FQDNSetExists(ctx, &sapb.FQDNSetExistsRequest{Domains: names})
+	if err != nil {
+		return fmt.Errorf("checking renewal exemption for %q: %s", names, err)
+	}
+	if exists.Exists {
+		ra.rateLimitCounter.WithLabelValues("new_order_by_registration_id", "FQDN set bypass").Inc()
```

Comment:
Adding the metrics to `//ra/ra_test.go`, I'm not yet understanding why a metric is incremented after the initial order is created. The diff below is currently passing unit tests fwiw.
```
diff --git a/ra/ra_test.go b/ra/ra_test.go
index 424ea2816..87e7c3fd4 100644
--- a/ra/ra_test.go
+++ b/ra/ra_test.go
@@ -964,8 +964,10 @@ func TestNewOrderRateLimiting(t *testing.T) {
        }
 
        // To start, it should be possible to create a new order
+       test.AssertMetricWithLabelsEquals(t, ra.rateLimitCounter, prometheus.Labels{"new_order_by_registration_id": "FQDN set bypass"}, 0)
        _, err := ra.NewOrder(ctx, orderOne)
        test.AssertNotError(t, err, "NewOrder for orderOne failed")
+       test.AssertMetricWithLabelsEquals(t, ra.rateLimitCounter, prometheus.Labels{"new_order_by_registration_id": "FQDN set bypass"}, 1)
 
        // Advance the clock 1s to separate the orders in time
        fc.Add(time.Second)
@@ -974,6 +976,7 @@ func TestNewOrderRateLimiting(t *testing.T) {
        // should fail
        _, err = ra.NewOrder(ctx, orderTwo)
        test.AssertError(t, err, "NewOrder for orderTwo succeeded, should have been ratelimited")
+       test.AssertMetricWithLabelsEquals(t, ra.rateLimitCounter, prometheus.Labels{"new_order_by_registration_id": "FQDN set bypass"}, 2)
```

Reply:
Disregard. I don't know what I was thinking back then.